os/arch/arm/src/armv7-a/arm_assert.c

```diff
@@ -582,10 +567,36 @@ void up_assert(const uint8_t *filename, int lineno)
 		asserted_location = (uint32_t)user_assert_location;
 		user_assert_location = 0x0;
 	} else {
-		asserted_location = (uint32_t)kernel_assert_location;
+		ARCH_GET_RET_ADDRESS(asserted_location)
```

Comment:
In this place, is this working? As far as I know, if we can this at middle of a function, it does not give a proper address. If we want to move this here, then I think we should use __builtin_..... instead of ARCH_GET....

Reply:
Right.. i reverted it. Thank you.